### PR TITLE
Sdk persistence of messages under project directory (MESDK-64)

### PR DESCRIPTION
**Pull request by jldec**
Resolves https://github.com/opral/inlang-message-sdk/issues/28

Messages are persisted in a raw JSON array of Message objects, in one file in `{projectname}.inlang/messages.json` 

For a small working project example see [sdk/multi-project-test/project4-dir](https://github.com/opral/monorepo/tree/242497d4cf6a0a1bfccea13611590a4c8d4678d2/inlang/source-code/sdk/multi-project-test/project4-dir) e.g. [translated JSON](https://github.com/opral/monorepo/blob/242497d4cf6a0a1bfccea13611590a4c8d4678d2/inlang/source-code/sdk/multi-project-test/project4-dir/messages.json.translated)

[sdk/load-test](https://github.com/opral/monorepo/tree/242497d4cf6a0a1bfccea13611590a4c8d4678d2/inlang/source-code/sdk/load-test) has been modified to generate fake messages using this persistence format. Early tests did not show significant aberrant behavior relative to plugin persistence. 

Further load testing with sherlock and fink is recommended, especialy since [normalization](https://github.com/opral/monorepo/blob/242497d4cf6a0a1bfccea13611590a4c8d4678d2/inlang/source-code/sdk/src/storage/helper.ts#L30) during serialization is expected to impact save performance. 

To dump messages from an existing project, the inlang validate CLI has been enhanced with an experimental --dumpFile option. This can be used for migration or testing with existing projects e.g.

```txt
~/opral/monorepo$ INLANG_CLI_EXPERIMENTAL=1 pnpm inlang validate --dumpFile messages.json --project project.inlang
ℹ 🔎 Validating the inlang project...
Using existing cloned repo
✔ Opened project at /Users/jldec/opral/monorepo/project.inlang
✔ Dumped messages JSON to /Users/jldec/opral/monorepo/messages.json
✔ The project is valid!     

# now copy the message.json file into project.inlang and turn on the feature flag project setting `experimental: { persistence: true }`.
```

- [x] Feature flagged under project settings `experimental: { persistence: true }`
- [x] override load/store plugin from project settings with sdk built-in persistence
- [x] uses the same loadMessages and saveMessages API as other plugins (for now) 
- [x] stringifies full message structure in non-human-editable JSON
- [x] also works if no load/store plugin is provided
- [x] cleanup load/store resolvedPluginApi parameter passing in loadProject()
- [x] cleanup unnecessary wrappers for loadMessages and saveMessages in resolvePlugins.ts 
- [x] add multi-project-test and modify load-test 
- [x] define migration requirements - see MESDK-74 (https://github.com/opral/inlang-message-sdk/issues/37)
- [x] plugin.ts api tests
- [x] changeset

### CODE REVIEW

File: inlang/source-code/sdk/src/resolve-modules/plugins/resolvePlugins.ts
Comment (samuelstroschein) [inline inlang/source-code/sdk/src/resolve-modules/plugins/resolvePlugins.ts:126]: can be turned into an if else statement

if (experimentalPersistence){

} else if()

File: inlang/source-code/sdk/src/persistence/plugin.ts
Comment (samuelstroschein) [inline inlang/source-code/sdk/src/persistence/plugin.ts:92]: The entire plugin logic is not required. 

The "persistence" can be hacked into: 

```ts
saveMessages(messages){
  JSON.stringify(messsages)
}
loadMessages(fs){
  JSON.parse(fs)
}

```

File: inlang/source-code/sdk/src/persistence/plugin.ts
Comment (samuelstroschein) [inline inlang/source-code/sdk/src/persistence/plugin.ts:18]: Remove the settings schema to avoid runtime validation errors

**Review by samuelstroschein — CHANGES_REQUESTED** [review]
LGTM but remove the plugin logic. Only two functions `loadMessages()` and `saveMessages()` are required for the feature flag approach

**Comment by jldec** [comment]
@samuelstroschein this should be ready to merge - if you want to take a final look.
@martin-lysk feel free to  👀 and offer feedback also. 
I will not merge before tomorrow morning.

**Review by samuelstroschein — APPROVED** [review]
Let's gooo. 

This is the easiest way to get persistence going. Have a feature flag and a persistency prototype by calling providing "AST dump," `loadMessages` and `saveMessages` functions. 

PS @martin-lysk this solution was the plan of @jldec all the time. The A bunch of miscommunication.

**Pull request merged**
